srcs/python/kungfu/tensorflow/optimizers/core.py

```diff
@@ -20,7 +20,24 @@ def defuse(y, shapes):
     return ts
 
 
-class KungFuOptimizer(tf.train.Optimizer):
+try:
```

Comment:
I think we can avoid `try` `catch` by determine the tf version from `tf.__version__`, e.g.

```python
_tf_major_version = int(tf.__version__.split('.')[0])

if _tf_major_version == 1: 
    ...
elif _tf_major_version == 2:
    ...
else:
   raise ... 

```